be/src/Project/Services/Client/Client.Api/Services/Groups/GroupsService.cs

```diff
@@ -0,0 +1,213 @@
+                if (groupTranslations is null)
+                {
+                    groupTranslations = this.context.GroupTranslations.FirstOrDefault(x => x.GroupId == group.Id && x.IsActive);
+                }
+
+                item.Name = groupTranslations?.Name;
+
+                groupItems.Add(item);
+            }
+
+            pagedGroupServiceModel.Data = groupItems;
+
+            return pagedGroupServiceModel;
+        }
+
+        public async Task<PagedResults<IEnumerable<GroupServiceModel>>> GetByIdsAsync(GetGroupsByIdsServiceModel model)
+        {
+            var groups = this.context.Groups.Where(x => model.Ids.Contains(x.Id) && x.IsActive);
+
+            if (string.IsNullOrWhiteSpace(model.SearchTerm) is false)
+            {
+                groups = groups.Where(x => x.Translations.Any(x => x.Name.StartsWith(model.SearchTerm)));
+            }
+
+            groups = groups.ApplySort(model.OrderBy);
+
+            var pagedResults = groups.PagedIndex(new Pagination(groups.Count(), model.ItemsPerPage), model.PageIndex);
+
+            var pagedGroupServiceModel = new PagedResults<IEnumerable<GroupServiceModel>>(pagedResults.Total, pagedResults.PageSize);
+
+            var groupItems = new List<GroupServiceModel>();
+
+            foreach (var group in pagedResults.Data.OrEmptyIfNull().ToList())
+            {
+                var item = new GroupServiceModel
+                {
+                    Id = group.Id,
+                    LastModifiedDate = group.LastModifiedDate,
+                    CreatedDate = group.CreatedDate
+                };
+
+                var groupTranslations = await this.context.GroupTranslations.FirstOrDefaultAsync(x => x.Language == model.Language && x.GroupId == group.Id && x.IsActive);
+                if (groupTranslations is null)
+                {
+                    groupTranslations = this.context.GroupTranslations.FirstOrDefault(x => x.GroupId == group.Id && x.IsActive);
+                }
+
+                item.Name = groupTranslations?.Name;
+
+                groupItems.Add(item);
+            }
+
+            pagedGroupServiceModel.Data = groupItems;
+
+            return pagedGroupServiceModel;
+        }
+
+        public async Task<Guid> UpdateAsync(UpdateGroupServiceModel model)
+        {
+            var group = await this.context.Groups.FirstOrDefaultAsync(x => x.Id == model.Id && x.IsActive);
+            
+            if (group == null)
+            {
+                throw new CustomException(this.clientLocalizer.GetString("GroupNotFound"), (int)HttpStatusCode.NotFound);
+            }
+
+            var groupTranslation = await this.context.GroupTranslations.FirstOrDefaultAsync(x => x.GroupId == model.Id && x.Language == model.Language && x.IsActive);
+
+            if (groupTranslation is not null)
+            {
+                groupTranslation.Name = model.Name;
+                groupTranslation.LastModifiedDate = DateTime.UtcNow;
+            } else
```

Comment:
**suggestion (non-blocking):** else should go to a new line. It's by convention in our C# code